Firmware/SpoolJoin.cpp

```diff
@@ -0,0 +1,80 @@
+#include "SpoolJoin.h"
+#include "Marlin.h"
+#include "eeprom.h"
+
+namespace SpoolJoin {
+
+SpoolJoin spooljoin;
+
+SpoolJoin::SpoolJoin()
+    : status(EEPROM::Unknown)
+    , currentMMUSlot(0)
+{
+}
+
+void SpoolJoin::updateSpoolJoinStatus(EEPROM newStatus)
+{
+    status = newStatus;
+    eeprom_write_byte((uint8_t*)EEPROM_AUTO_DEPLETE, (uint8_t)status);
+}
+
+void SpoolJoin::initSpoolJoinStatus()
+{
+    EEPROM currentStatus = (EEPROM)eeprom_read_byte((uint8_t*)EEPROM_AUTO_DEPLETE);
```

Comment:
another topic for potential optimization - many variables read from the eeprom are checked for unitialized values. We could try implementing a non-inline function:
`uint8_t eeprom_get_byte(uint8_t * address, uint8_t defaultValue)`

We could save some space doing the check for defaults at one spot and having always a valid value as a result. This function could be updating the eeprom cell with the default as well if necessary.